integration/token/interop/views/pledge/redeem.go

```diff
@@ -0,0 +1,77 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package pledge
+
+import (
+	"encoding/json"
+
+	view2 "github.com/hyperledger-labs/fabric-smart-client/platform/view"
+	"github.com/hyperledger-labs/fabric-smart-client/platform/view/services/assert"
+	"github.com/hyperledger-labs/fabric-smart-client/platform/view/view"
+	token2 "github.com/hyperledger-labs/fabric-token-sdk/token"
+	"github.com/hyperledger-labs/fabric-token-sdk/token/services/interop/pledge"
+	"github.com/hyperledger-labs/fabric-token-sdk/token/services/ttx"
+	"github.com/hyperledger-labs/fabric-token-sdk/token/token"
+)
+
+// Redeem contains the input information for a redeem
+type Redeem struct {
+	TMSID token2.TMSID
+	// TokenID contains the identifier of the token to be redeemed
+	TokenID *token.ID
+}
+
+type RedeemView struct {
+	*Redeem
+}
+
+func (rv *RedeemView) Call(context view.Context) (interface{}, error) {
+	w, err := pledge.GetIssuerWallet(context, "")
+	assert.NoError(err, "failed to retrieve wallet of issuer during redeem")
+
+	wallet := pledge.NewIssuerWallet(context, w)
+	t, script, err := wallet.GetPledgedToken(rv.TokenID)
+	assert.NoError(err, "failed to retrieve pledged token during redeem")
+
+	// make sure token was in fact claimed in the other network
+	proof, err := pledge.RequestProofOfTokenWithMetadataExistence(context, rv.TokenID, rv.TMSID, script)
```

Comment:
Maybe we change the name of the method to `RequestProofOfExistenceOfTokenWithMetadata`?

Reply:
okay